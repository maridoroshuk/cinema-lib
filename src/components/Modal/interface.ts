import { ReactNode } from 'react';

export interface IModal {
  dataCy: string;
  children: ReactNode;
}
