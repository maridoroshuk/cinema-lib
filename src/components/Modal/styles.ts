import styled from 'styled-components';

export const Wrapper = styled.div`
  position: relative;
  width: 100%;
  max-width: ${({ theme }) => theme.sizes.width[850]}px;
  padding: ${({ theme }) => theme.spaces[40]}px
    ${({ theme }) => theme.spaces[70]}px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: ${({ theme }) => theme.colors.primary800};
  z-index: ${({ theme }) => theme.zIndex[3000]};
`;
