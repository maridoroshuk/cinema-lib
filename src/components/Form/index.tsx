import React from 'react';
import { Formik } from 'formik';

import { IFormProps } from './interface';
import { StyledForm } from './styles';

export const Form = <T extends {}>({
  children,
  initialValues,
  validate,
  onFormSubmit,
}: IFormProps<T>) => {
  return (
    <Formik
      initialValues={initialValues}
      validate={validate}
      onSubmit={onFormSubmit}
    >
      <StyledForm>{children}</StyledForm>
    </Formik>
  );
};
