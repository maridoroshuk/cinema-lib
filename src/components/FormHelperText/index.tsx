import React from 'react';

import { IFormHelperTextProps } from './interface';
import { Wrapper } from './styles';

export const FormHelperText = ({ message }: IFormHelperTextProps) => {
  return <Wrapper>{message}</Wrapper>;
};
