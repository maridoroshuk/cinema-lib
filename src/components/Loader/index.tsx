import React from 'react';
import ReactLoading from 'react-loading';
import { ILoaderProps } from './interface';
import { LoaderWrapper } from './styles';

export function Loader({ height, width, color }: ILoaderProps) {
  return (
    <LoaderWrapper>
      <ReactLoading type="spokes" color={color} height={height} width={width} />
    </LoaderWrapper>
  );
}
