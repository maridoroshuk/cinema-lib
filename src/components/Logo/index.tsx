import React from 'react';

import LogoImage from '@/assets/Logo.png';
import { getLogoSize } from '@/helpers/getLogoSize';

import { LogoStyled } from './styles';
import { ILogo } from './interface';

export const Logo = ({ size }: ILogo) => {
  const width = getLogoSize(size);
  return (
    <LogoStyled src={LogoImage} width={width} alt="company logo (Modsen)" />
  );
};
