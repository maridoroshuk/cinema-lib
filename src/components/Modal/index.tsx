import React from 'react';

import { IModal } from './interface';
import { Wrapper } from './styles';

export const Modal = ({ children }: IModal) => {
  return <Wrapper>{children}</Wrapper>;
};
