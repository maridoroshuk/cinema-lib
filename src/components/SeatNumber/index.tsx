import React from 'react';

import { ISeatProps } from '@/components/SeatNumber/interface';
import { StyledSeat } from '@/components/SeatNumber/styles';

export const SeatNumber = ({
  children,
  backgroundColor,
  index,
}: Partial<ISeatProps>) => {
  return (
    <StyledSeat backgroundColor={backgroundColor} seatIndex={index}>
      {children}
    </StyledSeat>
  );
};
