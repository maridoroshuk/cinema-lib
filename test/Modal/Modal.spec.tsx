import React from 'react';
import { render } from '@testing-library/react';

import { Modal } from '@/components/Modal';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('Modal', () => {
  it('renders', () => {
    const { container } = render(
      setupTestProvider(
        <Modal>
          <p>Modal Text</p>
        </Modal>,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
