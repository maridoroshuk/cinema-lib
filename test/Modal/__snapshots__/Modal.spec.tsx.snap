// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
.c0 {
  position: relative;
  width: 100%;
  max-width: 850px;
  padding: 40px 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #1E1F27;
  z-index: 3000;
}

<div
  class="c0"
>
  <p>
    Modal Text
  </p>
</div>
`;
